### PR TITLE
Fix transaction type parsing

### PR DESCRIPTION
**Pull request by AnnaGolosova**
## Link to ClickUP ticket

https://app.clickup.com/t/1jvea9b

### *What is the problem we're trying to solve*
image

When we try to parse transaction model from FinancialTransactionsAPI, we got JSON serialization error

### *What changes have we introduced*

Add custom converter to convert TransactionType enum value from it's description to enum value
image

#### _Checklist_

- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: HousingSearchListener.Tests/V1/E2ETests/Stories/AddPersonToTenureTests.cs [hunk 1]
```diff
@@ -66,7 +66,6 @@ public void PersonNotFound()
                 .And(g => _personApiFixture.GivenThePersonDoesNotExist(_tenureApiFixture.AddedPersonId))
                 .When(w => _steps.WhenTheFunctionIsTriggered(tenureId, _tenureApiFixture.MessageEventData, EventTypes.PersonAddedToTenureEvent))
                 .Then(t => _steps.ThenTheCorrelationIdWasUsedInTheApiCall(_tenureApiFixture.ReceivedCorrelationIds))
-                .Then(t => _steps.ThenTheCorrelationIdWasUsedInTheApiCall(_personApiFixture.ReceivedCorrelationIds))
```
Comment (LBHMGeorgieva) [inline HousingSearchListener.Tests/V1/E2ETests/Stories/AddPersonToTenureTests.cs:69]: Why is this step removed?

File: HousingSearchListener/V1/UseCase/AddPersonToTenureUseCase.cs [hunk 2]
```diff
@@ -112,7 +112,7 @@ private static List<HouseholdMembers> GetHouseholdMembersFromEventData(object da
 
         private static T ConvertFromObject<T>(object obj) where T : class
         {
-            return JsonSerializer.Deserialize<T>(JsonSerializer.Serialize(obj), JsonOptions.CreateJsonOptions());
+            return JsonSerializer.Deserialize<T>(obj.ToString(), JsonOptions.CreateJsonOptions());
```
Comment (LBHMGeorgieva) [inline HousingSearchListener/V1/UseCase/AddPersonToTenureUseCase.cs:115]: What is the reason behind changing this?

**Review by LBHMGeorgieva — COMMENTED** [review]
Hi Hanna, can you please provide some description in the PR to understand the context?

File: HousingSearchListener/V1/UseCase/AddPersonToTenureUseCase.cs (hunk 2 above)
Comment (AnnaGolosova) [reply HousingSearchListener/V1/UseCase/AddPersonToTenureUseCase.cs]: There was issue in other palce, already rolled back this change

File: HousingSearchListener.Tests/V1/E2ETests/Stories/AddPersonToTenureTests.cs (hunk 1 above)
Comment (AnnaGolosova) [reply HousingSearchListener.Tests/V1/E2ETests/Stories/AddPersonToTenureTests.cs]: There was issue in other palce, already rolled back this change

File: HousingSearchListener/V1/Gateway/ApiGateway.cs [hunk 3]
```diff
@@ -84,7 +85,7 @@ public void Initialise(string apiName, string configKeyApiUrl, string configKeyA
 
             var responseBody = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
             if (response.IsSuccessStatusCode)
-                return JsonSerializer.Deserialize<T>(responseBody, _jsonOptions);
```
Comment (humulla) [inline HousingSearchListener/V1/Gateway/ApiGateway.cs:87]: Why are we changing it to `DeserializeObject`, this is a common class that is been used for all events

File: HousingSearchListener/V1/Gateway/ApiGateway.cs [hunk 4]
```diff
@@ -84,7 +85,7 @@ public void Initialise(string apiName, string configKeyApiUrl, string configKeyA
```
Comment (humulla) [inline HousingSearchListener/V1/Gateway/ApiGateway.cs:85]: We should remove this class and use Hackney.Core.Http nuget package

File: .circleci/config.yml [hunk 5]
```diff
@@ -160,12 +160,24 @@ jobs:
           aws-account: $AWS_ACCOUNT_PRODUCTION
 
 workflows:
+  check-and-build:
```
[the remark this answers is not in the document]
Comment (AnnaGolosova) [reply .circleci/config.yml:163]: Done

File: .circleci/config.yml [hunk 6]
```diff
@@ -160,12 +160,24 @@ jobs:
...
   check-and-deploy-development:
     jobs:
       - check-code-formatting:
           context: api-nuget-token-context
+          filters:
+            branches:
+              only: master
       - build-and-test:
           context: api-nuget-token-context
+          filters:
```
[the remark this answers is not in the document]
Comment (AnnaGolosova) [reply .circleci/config.yml:178]: Done

File: .circleci/config.yml [hunk 7]
```diff
@@ -160,12 +160,24 @@ jobs:
...
+    jobs:
+      - check-code-formatting:
+          context: api-nuget-token-context
+      - build-and-test:
+          context: api-nuget-token-context
   check-and-deploy-development:
     jobs:
       - check-code-formatting:
           context: api-nuget-token-context
+          filters:
```
[the remark this answers is not in the document]
Comment (AnnaGolosova) [reply .circleci/config.yml:173]: Done

**Comment by AnnaGolosova** [comment]
@humulla Hi! Thanks you for review! 
I have a question about `ApiGateway` from `Hackney.Core.Http` package.
As I can see, it uses private variable to store json convertation options.
But I need to set up new custom converter for `TransactionType` enum for Transactions model.

So, I don't have any ways to register this custom converter in `_jsonOptions`. That's why i changed serialization library to Newtonsoft.JSON

**Comment by humulla** [comment]
> @humulla Hi! Thanks you for review! I have a question about `ApiGateway` from `Hackney.Core.Http` package. As I can see, it uses private variable to store json convertation options. But I need to set up new custom converter for `TransactionType` enum for Transactions model.
> 
> So, I don't have any ways to register this custom converter in `_jsonOptions`. That's why i changed serialization library to Newtonsoft.JSON

Hi, I guess that's fine for your scenario, however that might cause issues for the other events, I would recommend for you to use the nuget package and then for the transaction model create a new serialization convertor

**Comment by AnnaGolosova** [comment]
@humulla All done by your last comment. I installed `Hackney.Core.Http` package and changed handlers to use it for all `ApiGateways` except `TransactionsApiGateway`. Can you please review again?

**Review by humulla — APPROVED** [review]

**Pull request merged**